### DISCUSSION TITLE
openEMS python interface

### DISCUSSION THREAD

**Discussion by warsea94** [thliebig/openEMS-Project#120] [Q&A Usage] [answered]
I have installed openEMS by following the instructions given [here](https://docs.openems.de/install/manual.html) at location ~/openEMS.
I am trying to Install Python bindings according to the instructions given in the same page but, getting an error 
```
/usr/bin/ld: cannot find -lCSXCAD
collect2: error: ld returned 1 exit status
error: command '/usr/bin/x86_64-linux-gnu-g++' failed with exit code 1
```
I have also tried using `python3 setup.py build_ext -I/openEMS/include -L/openEMS/lib -R/openEMS/lib` and even tried setting up `LD_LIBRARY_PATH=~/openEMS/lib` command but, unfortunately  still end up with the same error.
Any help will be much appreciated. 
Thanks in advanvce.

**Comment by thliebig** [comment, marked as answer]
If you have installed to `~/openEMS`, the path all showing towards `/openEMS` are not correct?
I also think that for the -I and -L etc options you should expand to the full path instead of `"~"`

**Reply by thliebig** [reply]
Btw, if you build with the script like this: `./update_openEMS.sh ~/openEMS --python` , the python interface is build as well...

**Reply by warsea94** [reply]
It works now. I added the full path to -I and -L options. Thank you

I proceeded to install the AppCSXCAD GUI and while installing i get the following errors:
1.
```
/openEMS-Project/QCSXCAD/QCSPrimEditor.cpp:44:16: error: ‘QString’ is not a namespace or unscoped enum
   44 | using QString::SkipEmptyParts;
```
2.
```
/openEMS-Project/QCSXCAD/QCSPrimEditor.cpp: In member function ‘virtual void QCSPrimPolygonLayout::SetValues()’:
/openEMS-Project/QCSXCAD/QCSPrimEditor.cpp:601:54: error: ‘SkipEmptyParts’ was not declared in this scope
  601 |  QStringList strListX = CoordLineX->text().split(",",SkipEmptyParts);
```

**Reply by thliebig** [reply]
Update the sources and try again, it was a Qt compat issue...
